### PR TITLE
[#175594485] Fix healthcheck gate deploy pipeline

### PR DESCRIPTION
**Pull request by pasqualedevita**
- add new parameter **containerInstanceResourceGroup**
- add **checks** if exists target vnet and subnet to avoid to create new vnet or subnet

### CODE REVIEW

File: azure-templates/deploy-steps.yml [hunk 1]
```diff
@@ -135,9 +140,27 @@ steps:
...
+          # check if vnet exists in target resource group (containerInstanceResourceGroup)
+          OUTPUT_VNET=$(az network vnet list --resource-group '${{ parameters.containerInstanceResourceGroup }}' --query "[?contains(name, '${{ parameters.containerInstanceVNet }}')].name" --output tsv)
+          if [ "$OUTPUT_VNET" = '${{ parameters.containerInstanceVNet }}' ]; then
+            echo "VNET '${{ parameters.containerInstanceVNet }}' found, continue"
+          else
+            echo "VNET '${{ parameters.containerInstanceVNet }}' not found, fail"
+            exit 1
+          fi
+
+          # check if subet exists in target resource group and vnet (containerInstanceResourceGroup, containerInstanceVNet)
```
Comment (gunzip) [inline azure-templates/deploy-steps.yml:152]:
```suggestion
          # check if subnet exists in target resource group and vnet (containerInstanceResourceGroup, containerInstanceVNet)
```

**Review by gunzip — APPROVED** [review]
lgtm cc @balanza

File: azure-templates/deploy-steps.yml [hunk 2]
```diff
@@ -37,6 +37,11 @@ parameters:
     type: string
     default: '0.5'
 
+  # the resource group must be the same where vnet is created
+  - name: 'containerInstanceResourceGroup'
+    type: string
+    default: 'io-p-rg-common'
```
Comment (balanza) [inline azure-templates/deploy-steps.yml:43]: Can't we get this value from the environment?

**Review by balanza — CHANGES_REQUESTED** [review]

File: azure-templates/deploy-steps.yml (hunk 1 above)
Comment (pasqualedevita) [reply azure-templates/deploy-steps.yml]: thanks

File: azure-templates/deploy-steps.yml (hunk 2 above)
Comment (pasqualedevita) [reply azure-templates/deploy-steps.yml:43]: `az network vnet subnet list` requires resource group to list subnets, so we can't get values from other resources.

I don't think that we'll changes these values often.

```
  # the resource group must be the same where vnet is created
  - name: 'containerInstanceResourceGroup'
    type: string
    default: 'io-p-rg-common'
  # attached vnet to the container instance
  - name: 'containerInstanceVNet'
    type: string
    default: 'io-p-vnet-common'
  # container instance subnet
  - name: 'containerInstanceSubnet'
    type: string
    default: 'azure-devops'
```

**Pull request merged**